sdk/core/src/policies/transport.rs

```diff
@@ -0,0 +1,49 @@
+use crate::policies::{Policy, PolicyResult};
+use crate::{BoxedFuture, Context, Request, Response};
+use std::pin::Pin;
+use std::sync::Mutex;
+
+type Transport = dyn Fn(&mut Context, Request) -> Pin<BoxedFuture<Response>> + Send;
```

Comment:
I do wonder if modeling this as a closure is the right thing to do. Another possibility is to have a trait like `Transport` which would have one method `send` on it. This might make implementation easier than a closure. Not sure...

Reply:
I am mulling over it as well. 
Closures are very succinct but can quickly become complicated to reason with. We'll probably have to reiterate more on this.